baybe/simulation/_imputation.py

```diff
@@ -0,0 +1,87 @@
+"""Target value imputation functionality."""
+
+from typing import Literal
+
+import numpy as np
+import pandas as pd
+
+from baybe.targets import NumericalTarget
+from baybe.targets.enum import TargetMode
+
+
+def _impute_lookup(
+    row: pd.Series,
+    lookup: pd.DataFrame,
+    targets: list[NumericalTarget],
```

Comment:
To be clear: it's not about what I "prefer" but what the internal code logic imposes as requirements on the collection, so it always needs to be decided case by case. This overview here might help:
https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

For the specific example you showed: note that I did NOT yet change/improve any of the type hints or similar, simply because we said that we want to keep the refactoring PRs as small as possible (remember the big PR I originally pushed?). So that will come in a second step 👍🏼

Reply:
ok but the PR desc already talks about rewriting docstrings and not just moving files so I thought the type ints (which are strongly coupled to the docstrings?) are potentially part of this PR

Reply:
Ok, I see. But no, was not touched. Will do in the next PR. And I know these things are often coupled, but typically only because of bad wording. For example, you'd have the coupling when you write "List of parameters", which I agree can make the user think that stuff needs to be entered as list, even though it can be an arbitrary collection instead. That's why I think we should always try to avoid such phrases, if possible.